dashboard/app/handler.go

```diff
@@ -78,6 +79,37 @@ func handleContext(fn contextHandler) http.Handler {
 	})
 }
 
+func isRobot(r *http.Request) bool {
+	userAgent := strings.ToLower(strings.Join(r.Header["User-Agent"], " "))
+	if strings.HasPrefix(userAgent, "curl") ||
+		strings.HasPrefix(userAgent, "wget") {
+		return true
+	}
+	return false
+}
+
+func backpressureRobots(c context.Context, r *http.Request) func() {
+	if !isRobot(r) {
+		return func() {}
+	}
+	cfg := getConfig(c).Throttle
+	if cfg.Empty() {
+		return func() {}
+	}
+	minDelay := cfg.Window.Milliseconds() / int64(cfg.Limit)
+	startTime := time.Now()
+	return func() {
+		actualDelay := time.Since(startTime).Milliseconds()
```

Comment:
This may be shorter w/o casts to int milliseconds by computing using Time/Duration:
```
	delayUntil := time.Now().Add(Window / time.Duration(Limit))
    ...
	<-time.After(time.Until(delayUntil))
```

Reply:
Nice. Thank you.